apps/emqx/src/emqx_app.erl

```diff
@@ -62,11 +61,11 @@ stop(_State) -> ok.
 %% @doc Call this function to make emqx boot without loading config,
 %% in case we want to delegate the config load to a higher level app
 %% which manages emqx app.
-set_init_config_load_done() ->
-    application:set_env(emqx, init_config_load_done, true).
+set_config_loader_prio(Priority) when is_integer(Priority), Priority >= 0 ->
```

Comment:
Nit (m.b. follow up):
Seems that the party setting the Prio always knows who is to populate the config. 
M.b. just set the concrete app explicitly?
```
emqx_app:set_config_loader(emqx_conf),
...
emqx_app:set_config_loader(emqx), 
...
```